Turn off AirPlay for unsupported cases.

megangardner opened the pull request:
#### 5d5f4ec1123380b50a799a61be248247f91cc750
<pre>
Turn off AirPlay for unsupported cases.
<a href="https://bugs.webkit.org/show_bug.cgi?id=252275">https://bugs.webkit.org/show_bug.cgi?id=252275</a>
rdar://103322594

Reviewed by Tim Horton.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WebCore/Modules/airplay/WebKitPlaybackTargetAvailabilityEvent.cpp:
* Source/WebCore/Modules/airplay/WebKitPlaybackTargetAvailabilityEvent.h:
* Source/WebCore/Modules/airplay/WebKitPlaybackTargetAvailabilityEvent.idl:
* Source/WebCore/dom/EventNames.in:
* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::enqueuePlaybackTargetAvailabilityChangedEvent):
* Source/WebCore/html/HTMLMediaElement.idl:

Canonical link: <a href="https://commits.webkit.org/260341@main">https://commits.webkit.org/260341@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e22d188ecc15ca70813eb8b08b6634ac5b34da1e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/108049 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/17112 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/40923 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/117176 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/116508 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/111937 "Passed tests") | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/76/builds/18480 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/8418 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/100247 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/113815 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/76/builds/18480 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/97166 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/41873 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/76/builds/18480 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/28804 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/83545 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [  ~~🛠 🧪 jsc~~](https://ews-build.webkit.org/#/builders/1/builds/97281 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/81/builds/10009 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/70/builds/30152 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 jsc-armv7~~](https://ews-build.webkit.org/#/builders/38/builds/96651 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/83/builds/8099 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/10728 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/7051 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/38/builds/96651 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/80/builds/16159 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/62/builds/49743 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 jsc-mips~~](https://ews-build.webkit.org/#/builders/37/builds/105664 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7156 "Failed to push commit to Webkit repository") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/12307 "Built successfully") | | [  ~~🧪 jsc-mips-tests~~](https://ews-build.webkit.org/#/builders/37/builds/105664 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | | | | 
<!--EWS-Status-Bubble-End-->

hortont424 commented on Source/WTF/wtf/PlatformEnableCocoa.h:
```diff
@@ -758,6 +758,12 @@
+#define ENABLE_WIRELESS_PLAYBACK_TARGET_AVALABILITY_API 1
+#endif
+
+
```
Extra new lines

hortont424 commented on Source/WebCore/Modules/airplay/WebKitPlaybackTargetAvailabilityEvent.cpp:
```diff
@@ -29,7 +29,7 @@
 #include <wtf/NeverDestroyed.h>
 
-#if ENABLE(WIRELESS_PLAYBACK_TARGET)
+#if ENABLE(ENABLE_WIRELESS_PLAYBACK_TARGET_AVALABILITY_API)
```
ENABLE(ENABLE is wrong

hortont424 commented on Source/WebCore/Modules/airplay/WebKitPlaybackTargetAvailabilityEvent.idl:
```diff
@@ -25,7 +25,7 @@
 
 [
-    Conditional=WIRELESS_PLAYBACK_TARGET,
+    Conditional=ENABLE_WIRELESS_PLAYBACK_TARGET_AVALABILITY_API,
```
I think the ENABLE_ here is wrong but check the generated code

hortont424 commented on Source/WTF/wtf/PlatformEnableCocoa.h:
```diff
@@ -758,6 +758,10 @@
 #define ENABLE_WIRELESS_PLAYBACK_TARGET 1
 #endif
 
+#if !defined(ENABLE_WIRELESS_PLAYBACK_TARGET_AVALABILITY_API) && ENABLE_WIRELESS_PLAYBACK_TARGET && !HAVE(UIKIT_WEBKIT_INTERNALS)
```
this should be ENABLE(WIRELESS_PLAYBACK_TARGET) not ENABLE_

hortont424 commented on Source/WTF/wtf/PlatformEnableCocoa.h:
```diff
@@ -758,6 +758,10 @@
 #endif
 
+#if !defined(ENABLE_WIRELESS_PLAYBACK_TARGET_AVALABILITY_API) && ENABLE_WIRELESS_PLAYBACK_TARGET && !HAVE(UIKIT_WEBKIT_INTERNALS)
+#define ENABLE_WIRELESS_PLAYBACK_TARGET_AVALABILITY_API 1
```
AVALABILITY -> AVAILABILITY

megangardner committed:
Turn off AirPlay for unsupported cases.
https://bugs.webkit.org/show_bug.cgi?id=252275
rdar://103322594

Reviewed by Tim Horton.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WTF/wtf/PlatformEnableCocoa.h:
* Source/WebCore/Modules/airplay/WebKitPlaybackTargetAvailabilityEvent.cpp:
* Source/WebCore/Modules/airplay/WebKitPlaybackTargetAvailabilityEvent.h:
* Source/WebCore/Modules/airplay/WebKitPlaybackTargetAvailabilityEvent.idl:
* Source/WebCore/dom/EventNames.in:
* Source/WebCore/html/HTMLMediaElement.cpp:
(WebCore::HTMLMediaElement::enqueuePlaybackTargetAvailabilityChangedEvent):
* Source/WebCore/html/HTMLMediaElement.idl:

Canonical link: https://commits.webkit.org/260341@main
